src/commands/client/upsert/index.ts

```diff
@@ -1,11 +1,12 @@
-import type { NAMESPACE } from "@commands/client/types";
+import type { NAMESPACE, SparseVector } from "@commands/client/types";
 import { Command } from "@commands/command";
 
 type NoInfer<T> = T extends infer U ? U : never;
 
 type VectorPayload<TMetadata> = {
```

Comment:
This type allows both vector and sparseVector not be passed. At least one should be passed in type